ldap/servers/slapd/daemon.c

```diff
@@ -918,6 +920,41 @@ slapd_sockets_ports_free(daemon_ports_t *ports_info)
 #endif
 }
 
+/*
+ * slapi_eq_repeat_rel callback that checks that idletimeout has not expired.
+ */
+void
+check_idletimeout(time_t when __attribute__((unused)), void *arg __attribute__((unused)) )
+{
+    Connection_Table *ct = the_connection_table;
+    time_t curtime = slapi_current_rel_time_t();
+    /* Walk all active connections of all connection listeners */
+    for (int list_num = 0; list_num < ct->list_num; list_num++) {
+        for (Connection *c = connection_table_get_first_active_connection(ct, list_num);
+             c != NULL; c = connection_table_get_next_active_connection(ct, c)) {
+            if (c->c_state == CONN_STATE_FREE || c->c_gettingber ||
+                c->c_ops != NULL || c->c_idletimeout <= 0 ||
+                curtime - c->c_idlesince < c->c_idletimeout) {
+                continue;
+            }
+            /* Looks like idletimeout has expired, lets acquire the lock
+             * and double check.
+             */
+            if (pthread_mutex_trylock(&(c->c_mutex)) == EBUSY) {
+                continue;
+            }
+            if (c->c_state != CONN_STATE_FREE && !c->c_gettingber &&
```

Comment:
This test is looking the opposite (!) of the test above (that skip this connection). It would be good to have a single macro/function.

Reply:
Added an inline function